custom-instrumentation/instrument-datastore/instrumentation.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const newrelic = require('newrelic')
+const datastorePath = require.resolve('./simple-datastore')
+
+newrelic.instrumentDatastore({
+  // The path to our datastore module
+  absolutePath: datastorePath,
+  // The name of the datastore module
+  moduleName: 'simple-datastore',
+  // The function that will instrument the datastore
+  // if it is successfully required.
+  onRequire: instrumentSimpleDatastore,
+  // The function that will be called if there is
+  // an error with the instrumentation.
+  onError: function myErrorHandler (err) {
+    console.error(err.message, err.stack)
+    process.exit(1)
+  },
+})
+
+function instrumentSimpleDatastore (shim, datastore, moduleName) {
+  console.log(`[NEWRELIC] instrumenting ${moduleName}`)
+  const proto = datastore.SimpleDatastore.prototype
```

Comment:
By unwrapping the export, we can directly work with the exported object:

```suggestion
function instrumentSimpleDatastore (shim, SimpleDatastore, moduleName) {
  console.log(`[NEWRELIC] instrumenting ${moduleName}`)
  const proto = SimpleDatastore.prototype
```